test/unit/testExecutorCurrentNameReason.js

```diff
@@ -0,0 +1,241 @@
+
+describe('ExecutorCurrentNameReason', () => {
+    const steps = initSteps([__dirname + '/../../app/steps/action/', __dirname + '/../../app/steps/ui']);
+
+    describe('getUrl()', () => {
+        it('returns the url with a * param when no index is given', (done) => {
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const url = ExecutorCurrentNameReason.constructor.getUrl();
+
+            expect(url).to.equal(`${executorCurrentNameReasonPath}*`);
+            done();
+        });
+
+        it('returns the url with the index as a param when an index is given', (done) => {
+            const param = 1;
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const url = ExecutorCurrentNameReason.constructor.getUrl(param);
+
+            expect(url).to.equal(executorCurrentNameReasonPath + param);
+            done();
+        });
+    });
+
+    describe('getContextData()', () => {
+        it('sets the index when there is a numeric url param', (done) => {
+            const req = {
+                session: {
+                    form: {
+                        executors: {
+                            list: []
+                        }
+                    }
+                },
+                params: [1]
+            };
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const ctx = ExecutorCurrentNameReason.getContextData(req);
+
+            expect(ctx.index).to.equal(req.params[0]);
+            done();
+        });
+
+        it('sets otherExecName', (done) => {
+            const req = {
+                params: {
+                    param: '*'
+                },
+                session: {
+                    form: {
+                        executors: {
+                            list: [
+                                {currentName: 'executor current name', hasOtherName: true},
+                                {currentName: 'bob smith', hasOtherName: true}
+                            ],
+                            index: 0
+                        }
+                    }
+                }
+            };
+            const executors = req.session.form.executors;
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const ctx = ExecutorCurrentNameReason.getContextData(req);
+
+            expect(ctx.otherExecName).to.equal(executors.list[0].currentName);
+            done();
+        });
+    });
+
+    describe('handleGet()', () => {
+        let testCtx;
+
+        beforeEach(() => {
+            testCtx = {
+                list: [{
+                    isApplying: true
+                }, {
+                    isApplying: true,
+                    currentNameReason: 'marriage'
+                }],
+                index: 1
+            };
+        });
+
+        it('returns the ctx with currentNameReason', (done) => {
```

Comment:
This test appears to be a duplicate of the one below.

There is only one positive scenario in this method so we should only need one test to test the returned params.